polycommit/src/sonic_pc/mod.rs

```diff
@@ -94,6 +94,80 @@ impl<E: PairingEngine> SonicKZG10<E> {
     }
 }
 
+macro_rules! execute_in_parallel {
```

Comment:
This is the core macro that we rely on. Essentially, the idea is that, if we have, say, n MSMs, and m threads, then instead of devoting all m threads to a single MSM, and then computing the MSMs one-by-one, we instead devote a smaller number (say, 12) of threads to each MSM, and perform m/12 MSMs in parallel. 

This results in speed ups because increasing the number of threads per MSM provides diminishing returns, so it makes sense to cap the number of threads for any given MSM, and use the leftover threads for other MSMs. (Sounds like the Robin Hood story lol)